hasher-matcher-actioner/src/OpenMediaMatch/app.py

```diff
@@ -96,12 +101,26 @@ def create_app() -> flask.Flask:
 
     scheduler: APScheduler | None = None
 
+    if _is_gunicorn() and (
+        app.config.get("ROLE_HASHER", False)
+        or app.config.get("ROLE_MATCHER", False)
+        or app.config.get("ROLE_CURATOR", False)
+    ) and (
+        app.config.get("TASK_INDEXER", False)
+        or app.config.get("TASK_FETCHER", False)
+    ):
```

Comment:
blocking questions: Instead, should the check be against number of workers? These tasks are safe to run as long as long as there is only one thread.

I'm hesitant to add more invariants on the startup that might block reasonable setups.